openedx/core/lib/xblock_services/video_config_utils.py

```diff
@@ -0,0 +1,87 @@
+"""
+Video configuration utilities for XBlock runtime.
+
+This module provides utility functions for video-related configuration
+that are specific to the edx-platform implementation.
+"""
+
+import logging
+
+from django.conf import settings
+
+from openedx.core.djangoapps.video_config.toggles import PUBLIC_VIDEO_SHARE
+from openedx.core.lib.courses import get_course_by_id
+from xmodule.course_block import (
+    COURSE_VIDEO_SHARING_ALL_VIDEOS,
+    COURSE_VIDEO_SHARING_NONE,
+)
+
+log = logging.getLogger(__name__)
+
+
+def get_public_video_url(video_block):
```

Comment:
Unless I missed something, it seems like all method in this file could simply take the video's `usage_key` as an argument instead of the block itself. This would make the methods easier to reason about (less input data, less potential for mutation) and easier to test.